### DISCUSSION TITLE
Read CSV with comments

### DISCUSSION THREAD

**Discussion by jiangyinzuo** [duckdb/duckdb#9831] [Ideas]
Can we add a feature like the following to `read_csv` function?

Suppose we have a `foo.csv`
```csv
# This is a comment.
name,age
Alice,20
Bob,23
```

Execute
```sql
SELECT name FROM read_csv('foo.csv', comment='#');
```

The result would be

| name  |
|-------|
| Alice |
| Bob   |

**Comment by pdet** [comment]
It seems that Polars and Pandas seem to support that option, so I'm not opposed to implementing it.
However, the sniffer should be capable of automatically skipping comments on the top of a file. 

Is that not working for you?

**Reply by jiangyinzuo** [reply]
Oh, you are right.  In fact, my use case is 
```
name,age
# comment
Alice,20
Bob,23
```
and the `read_csv` function can not skip the comment right now.

**Reply by pdet** [reply]
The sniffer only detects dirty rows on top of the file, so indeed comments mid-data will be problematic.

 I think adding a comment option and sniffing it is perfectly fine, will have a look at it in the future.

**Reply by kfsone** [reply]
Supporting files with comments in would be really nice but may not be entirely trivial due to the number of dialects, for instance, the CSV files I have to work with have _both_ '#'_ and ';' comments, and one of them frustratingly uses C++ style '//', and we sections in the file like:

```
# This section should only be buildings.

BUILDING  # for Amsterdam
PLACEMENT,  ; auto
ZONE,xx,yy,nnn,nnn,nnn,...   ; , 123, if we need to re-enable static ids
```

(I'm quite aware that csv may be the wrong format for these files)

**Reply by pdet** [reply]
Different comments dialects in the same file? 
The DuckDB's CSV reader now supports comments, but only one comment character per file.
